### PR TITLE
feat(SplunkLogger): Add requestOptions parameter

### PR DESCRIPTION
**Pull request by elliotttf**
This will allow `requestOptions` to be passed into the constructor rather than requiring modifications after instantiation. This is particularly useful for assigning multiple options at once, e.g.

```javascript
const logger = new SplunkLogger(config, {
  agent: new https.Agent(),
  strictSSL: true,
  timeout: 150,
});
```

### CODE REVIEW

**Comment by shakeelmohamed** [comment]
We need to support non-es6 versions of Node. That's the only thing blocking me from merging this - however I can't promise a release anytime soon due to tight engineering resources

**Comment by elliotttf** [comment]
@shakeelmohamed I've re-rolled the commit to not use ES6 features. Let me know if you want me to change anything else.

File: splunklogger.js [hunk 1]
```diff
@@ -66,7 +71,7 @@ function _defaultEventFormatter(message, severity) {
  * var logger = new SplunkLogger(config);
  *
  * @property {object} config - Configuration settings for this <code>SplunkLogger</code> instance.
- * @param {object} requestOptions - Options to pass to <code>{@link https://github.com/request/request#requestpost|request.post()}</code>.
+ * @property {object} requestOptions - Options to pass to <code>{@link https://github.com/request/request#requestpost|request.post()}</code>.
```
Comment (shakeelmohamed) [inline splunklogger.js]: This is not both a param & a property, so we'll need a near duplicate line

**Comment by shakeelmohamed** [comment]
Thanks @elliotttf. I updated the jshint config, if you merge/rebase from the develop branch you'll see a couple indent issues to fix.

How urgently do you need a release with this change?

**Comment by elliotttf** [comment]
@shakeelmohamed fixed!

It's not urgent but definitely a nice to have. Specifically I'm interested in this so that I have finer grain control over the socket pool that my application uses to communicate with splunk's servers.

It's still possible to do that without this change, just not as convenient 😄

**Comment by ABastionOfSanity** [comment]
+1

**Comment by vinnieking06** [comment]
Any update on when this might get merged? Our team is patching the package locally for the time being. Thanks!

**Comment by vinnieking06** [comment]
@shakeelmohamed or any other maintainer, looks like this one is ready to go when you have a moment to review. Thank you!

**Comment by ncanumalla-splunk** [comment]
Hi all, Shakeel is no longer one of the maintainers of this project.
@ashah-splunk @bparmar-splunk Can we please plan to review this PR as one of the next items? Thanks.

**Comment by vinnieking06** [comment]
Hey splunk team (@ncanumalla-splunk, @ashah-splunk ), any update on when this might get merged? My team appreciates it!

**Pull request merged**